Assets/UI/WorldView/citybannermanager_CQUI_expansions.lua

```diff
@@ -0,0 +1,188 @@
+        return;
+    end
+
+    local currentPopulation:number = pCity:GetPopulation();
+    -- XP1+, grab the first instance
+    local populationInstance = CQUI_GetInstanceObject(self.m_StatPopulationIM);
+
+    -- Get real housing from improvements value
+    local localPlayerID = Game.GetLocalPlayer();
+
+    -- CQUI : housing left
+    if (IsCQUI_SmartBanner_PopulationEnabled()) then
+        local cqui_HousingFromImprovementsCalc = CQUI_GetRealHousingFromImprovementsValue(pCity, localPlayerID);
+        if (cqui_HousingFromImprovementsCalc ~= nil) then    -- CQUI real housing from improvements fix to show correct values when waiting for the next turn
+            local housingText, housingLeft = CQUI_GetHousingString(pCity, cqui_HousingFromImprovementsCalc, true);
+            populationInstance.CQUI_CityHousing:SetText(housingText);
+            populationInstance.CQUI_CityHousing:SetHide(false);
+
+            -- CQUI : add housing left to tooltip
+            local popTooltip = populationInstance.FillMeter:GetToolTipString();
+            local CQUI_housingLeftPopupText = "[NEWLINE] [ICON_Housing]" .. Locale.Lookup("LOC_HUD_CITY_HOUSING") .. ": " .. housingLeft;
+            popTooltip = popTooltip .. CQUI_housingLeftPopupText;
+            populationInstance.FillMeter:SetToolTipString(popTooltip);
+        end
+    else
+        populationInstance.CQUI_CityHousing:SetHide(true);
+    end
+end
+
+-- ============================================================================
+function CityBanner.UpdateStats(self)
+    print_debug("CityBannerManager_CQUI_Expansions: CityBanner.UpdateStats ENTRY");
+    BASE_CQUI_CityBanner_UpdateStats(self);
+
+    local pDistrict:table = self:GetDistrict();
+    if (pDistrict ~= nil and IsBannerTypeCityCenter(self.m_Type)) then
+        local localPlayerID:number = Game.GetLocalPlayer();
+        local pCity        :table  = self:GetCity();
+        local iCityOwner   :number = pCity:GetOwner();
+
+        if (localPlayerID == iCityOwner and self.CQUI_DistrictBuiltIM ~= nil) then
+            -- On first call into UpdateStats, CQUI_DistrictBuiltIM may not be instantiated yet
+            -- However this is called often enough that it's not a problem
+            -- Update the built districts 
+            self.CQUI_DistrictBuiltIM:ResetInstances(); -- CQUI : Reset CQUI_DistrictBuiltIM
+            local pCityDistricts:table = pCity:GetDistricts();
+            if (IsCQUI_SmartBanner_DistrictsEnabled()) then
+                for i, district in pCityDistricts:Members() do
+                    local districtType = district:GetType();
+                    local districtInfo:table = GameInfo.Districts[districtType];
+                    local isBuilt = pCityDistricts:HasDistrict(districtInfo.Index, true);
+                    if (isBuilt == true
+                        and districtInfo.Index ~= 0
+                        and districtInfo.DistrictType ~= "DISTRICT_GOVERNMENT"
```

Comment:
Hm? Why is Gov Plaza excluded? It is a pretty important district.

Reply:
I think I had it confused for the city center, good catch will fix

Reply:
Fixed.